src/server/session.ts

```diff
@@ -286,6 +286,16 @@ namespace ts.server {
             : deduplicate(outputs, areEqual);
     }
 
+    function combineProjectOutputFromEveryProject<T>(projectService: ProjectService, action: (project: Project) => ReadonlyArray<T>, getFileName: (value: T) => string) {
+        const outputs: T[] = [];
+        projectService.forEachProject(project => {
+            if (project.isOrphan() || !project.languageServiceEnabled) return;
+            const theseOutputs = action(project);
+            outputs.push(...theseOutputs.filter(output => !outputs.some(o => getFileName(o) === getFileName(output))));

```

Comment:
Instead of taking getFileName as input i think you want isSame/compare(output1: T, output2: T): boolean as input so that its more generic function since thats what determines if the output is unique and to be added to the result